streams/src/main/java/org/apache/kafka/streams/KafkaClientSupplier.java

```diff
@@ -63,4 +64,13 @@
      * @return an instance of Kafka consumer
      */
     Consumer<byte[], byte[]> getRestoreConsumer(final Map<String, Object> config);
+
+    /**
+     * Create a {@link Consumer} which is used to restore records for {@link GlobalKTable}.
+     *
+     * @param config {@link StreamsConfig#getGlobalConsumerConfigs(String) global consumer config} which is supplied
+     *               by the {@link StreamsConfig} given to the {@link KafkaStreams}
```

Comment:
I am just realizing that we deprecated the constructor that takes a `StreamsConfig` in favor of constructors taking `Properties` -- we should update the JavaDoc accordingly. Can we do this in the PR as a "side tasks"?

Reply:
This function is called in a private function in `KafkaStreams`, where the input config was indeed supplied by `StreamsConfig`. Could you give me an example of how we want to rephrase the java doc?

Reply:
Just replace `{@link StreamsConfig}` with `{@link java.utils.Properties Properties}` (it's not a typo -- we need to use the full package path because we cannot import the class because we only use it in JavaDocs but not in code -- findbugs would complain about an unused import -- the second "Properties" is the actual used "link text" to hide the fully qualified class name).